main.go

```diff
@@ -26,7 +26,6 @@ func main() {
 	setupFlags()
 
 	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
```

Comment:
We should cancel the context to ensure that all go routines get the signal when the program shuts down.

Reply:
Done.